clients/go/zbc/client.go

```diff
@@ -136,6 +137,38 @@ func NewZBClient(config *ZBClientConfig) (ZBClient, error) {
 	}, nil
 }
 
+// NewOAuthZBClient is a convenience function that builds a client with an OAuth credentials provider using the provided
+// credentials. The gateway address is used as the OAuth token's audience and a default value is used in place of the
+// authorization server URL. This function is meant to simplify client configuration for Camunda Cloud.
+func NewOAuthZBClient(gatewayAddress, clientID, clientSecret string) (ZBClient, error) {
+	return newOAuthZBClientWithAuthzURL(&ZBClientConfig{GatewayAddress: gatewayAddress}, clientID, clientSecret, "")
+}
+
+func newOAuthZBClientWithAuthzURL(config *ZBClientConfig, clientID, clientSecret, authzServer string) (ZBClient, error) {
+	const portRegex = ":[0-9]+$"
```

Comment:
Do we need to use regex? Can't we just check for something like `lastIndexOf(':')` or whatever is the equivalent in Go, and do a substring up to that index (exclusive)?

EDIT: I think https://golang.org/pkg/strings/#LastIndex would work